contracts/src/tests/test_erc677.cairo

```diff
@@ -54,11 +55,13 @@ fn setup_invalid_receiver() -> (ContractAddress, MockInvalidReceiverDispatcher)
     (address, contract)
 }
 
+type ComponentState = ERC677Component::ComponentState<chainlink::token::link_token::LinkToken::ContractState>;
+
 fn transfer_and_call(receiver: ContractAddress) {
     let data = ArrayTrait::<felt252>::new();
     // have to send 0 because ERC20 is not initialized with starting supply when using this library by itself
-    let mut erc677 = ERC677::unsafe_new_contract_state();
-    //  TODO: ERC677::transfer_and_call(ref erc677, receiver, u256 { high: 0, low: 0 }, data);
+    let mut state: ComponentState = ERC677Component::component_state_for_testing();
```

Comment:
nice! good to know how this is done

Reply:
It's in the book! https://book.cairo-lang.org/ch99-01-05-03-testing-components.html#testing-components-without-deploying-a-contract